compiler/src/dotty/tools/dotc/typer/Checking.scala

```diff
@@ -377,6 +377,12 @@ object Checking {
         fail(CannotHaveSameNameAs(sym, cls, CannotHaveSameNameAs.CannotBeOverridden))
         sym.setFlag(Private) // break the overriding relationship by making sym Private
       }
+    checkApplicable(UnusedType, !sym.is(UnusedType))
```

Comment:
This was a result of defining the `Unused` flag as a `commonFlag`, but it can be a `termFlag` which makes this test redundant. I copied that from the flag for `implicit`, I wonder if that one can ever be a type flag. I will check that.

Reply:
Oh, for implicits it is needed for `implicit class ...`